config/detekt/baseline.xml

```
@@ -0,0 +1,419 @@
+    <ID>SpreadOperator:UiHelpers.kt$UiHelpers$( uiString.stringRes, *uiString.params.map { value -&gt; getTextOfUiString( context, value ) }.toTypedArray() )</ID>
+    <ID>ThrowsCount:SectionedRecyclerViewAdapter.kt$SectionedRecyclerViewAdapter$ fun notifyNotLoadedStateChanged(section: Section, previousState: State)</ID>
+    <ID>ThrowsCount:SectionedRecyclerViewAdapter.kt$SectionedRecyclerViewAdapter$ fun notifyStateChangedToLoaded(section: Section, previousState: State)</ID>
+    <ID>TooGenericExceptionCaught:Address.kt$Address$e: NullPointerException</ID>
+    <ID>TooGenericExceptionCaught:AppPrefs.kt$AppPrefs$ex: Throwable</ID>
+    <ID>TooGenericExceptionCaught:DateExt.kt$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:DateUtils.kt$DateUtils$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:DateUtils.kt$DateUtils$e: IndexOutOfBoundsException</ID>
+    <ID>TooGenericExceptionCaught:FCMRegistrationIntentService.kt$FCMRegistrationIntentService$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:FileUtils.kt$FileUtils$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:OrderShipmentTrackingHelper.kt$OrderShipmentTrackingHelper$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:PhoneUtils.kt$PhoneUtils$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:ProductDetailViewModel.kt$ProductDetailViewModel$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:StringDateFormatExt.kt$e: Exception</ID>
+    <ID>TooGenericExceptionCaught:TokenProvider.kt$TokenProvider$e: RuntimeException</ID>
+    <ID>TooGenericExceptionCaught:WooLogViewerActivity.kt$WooLogViewerActivity$e: Exception</ID>
+    <ID>TooGenericExceptionThrown:ActionableEmptyView.kt$ActionableEmptyView$throw RuntimeException("$context: ActionableEmptyView must have a title (aevTitle)")</ID>
+    <ID>TooManyFunctions:AddAttributeFragment.kt$AddAttributeFragment : BaseProductFragment</ID>
```

I think this `TooManyFunctions` rule will get in our way to much, especially in Activities and Fragments where we need a big amount of functions by default.
I believe we should modify the [rule](https://detekt.github.io/detekt/complexity.html#toomanyfunctions) by enabling `ignoreOverridden`, which would ignore all the overridden function from the rule, and then re-calculate the baseline, WDYT?
cc @woocommerce/android-developers for your opinion as well

---

My plan was that a discussion like this will go to phase 2. I will create a P2 where we could comment on the cases like that based on our real day-to-day work. After some time (a few weeks) we will have a general overview of possible modifications we might need. Then based on the collective decisions we would ignore/modify the rules via the detekt.xml file. Should it work, wdyt?

---

I think this would work for most of the other rules, but for the rule `TooManFunctions` rule, once we enable the integration, and before getting a chance to discuss and agree on the rules, we may be blocked for any new fragment/activity that we add in the app, and we may need to suppress the error explicitly.
This is just my opinion 🙂, and it's not a big deal, if you think leaving it to later is better, then go for it 👍

---

If this will be really striking us in a foot super often then we can have a quick discussion in woo-async and then change configuration right in the PR which is blocked by this